### PR TITLE
Implemented Graph DS in Rust

### PR DESCRIPTION
**Pull request by barkincavdaroglu**
First part of converting modules to Rust from JS. This PR contains an implementation of a graph data structure that will be used by algorithms. For more info, please view the ticket that is linked to this PR.

### CODE REVIEW

File: src-tauri/src/graph.rs [hunk 1]
```diff
@@ -0,0 +1,317 @@
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug)]
```
Comment (ajmcquilkin) [inline src-tauri/src/graph.rs:3]: Is this something we need to keep in the codebase?

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -32,6 +34,7 @@ fn greet(name: &str) -> String {
 }
 
 fn main() {
+    // let mut g = Graph::new();
```
Comment (ajmcquilkin) [inline src-tauri/src/main.rs:37]: Nit: remove commented line

**Review by ajmcquilkin — APPROVED** [review]
Looks good! Just a few minor changes for code clarity

File: src-tauri/src/graph.rs (hunk 1 above)
Comment (barkincavdaroglu) [reply src-tauri/src/graph.rs:3]: We need this to see the output of print statements in tests. I think it's best to keep them for now and I can remove later.

**Review by ajmcquilkin — APPROVED** [review]

**Pull request merged**